cmd/telegraf/main.go

```diff
@@ -0,0 +1,356 @@
+	_ "github.com/influxdata/telegraf/plugins/processors/all"
+	"github.com/urfave/cli/v2"
+)
+
+type TelegrafConfig interface {
+	CollectDeprecationInfos([]string, []string, []string, []string) map[string][]config.PluginDeprecationInfo
+	PrintDeprecationList([]config.PluginDeprecationInfo)
+}
+
+type Filters struct {
+	section    []string
+	input      []string
+	output     []string
+	aggregator []string
+	processor  []string
+}
+
+func processFilterFlags(section, input, output, aggregator, processor string) Filters {
+	sectionFilters := deleteEmpty(strings.Split(section, ":"))
+	inputFilters := deleteEmpty(strings.Split(input, ":"))
+	outputFilters := deleteEmpty(strings.Split(output, ":"))
+	aggregatorFilters := deleteEmpty(strings.Split(aggregator, ":"))
+	processorFilters := deleteEmpty(strings.Split(processor, ":"))
+	return Filters{sectionFilters, inputFilters, outputFilters, aggregatorFilters, processorFilters}
+}
+
+func deleteEmpty(s []string) []string {
+	var r []string
+	for _, str := range s {
+		if str != "" {
+			r = append(r, str)
+		}
+	}
+	return r
+}
+
+// runApp defines all the subcommands and flags for Telegraf
+// this abstraction is used for testing, so outputBuffer and args can be changed
+func runApp(args []string, outputBuffer io.Writer, pprof Server, c TelegrafConfig, m App) error {
+	pluginFilterFlags := []cli.Flag{
+		&cli.StringFlag{
+			Name:  "section-filter",
+			Usage: "filter the sections to print, separator is ':'. Valid values are 'agent', 'global_tags', 'outputs', 'processors', 'aggregators' and 'inputs'",
+		},
+		&cli.StringFlag{
+			Name:  "input-filter",
+			Usage: "filter the inputs to enable, separator is :",
+		},
+		&cli.StringFlag{
+			Name:  "output-filter",
+			Usage: "filter the outputs to enable, separator is :",
+		},
+		&cli.StringFlag{
+			Name:  "aggregator-filter",
+			Usage: "filter the aggregators to enable, separator is :",
+		},
+		&cli.StringFlag{
+			Name:  "processor-filter",
+			Usage: "filter the processors to enable, separator is :",
+		},
+	}
+
+	extraFlags := append(pluginFilterFlags, cliFlags()...)
+
+	app := &cli.App{
+		Name:   "Telegraf",
+		Usage:  "The plugin-driven server agent for collecting & reporting metrics.",
+		Writer: outputBuffer,
+		Flags: append(
+			[]cli.Flag{
+				// String slice flags
+				&cli.StringSliceFlag{
+					Name:  "config",
+					Usage: "configuration file to load",
+				},
+				&cli.StringSliceFlag{
+					Name:  "config-directory",
+					Usage: "directory containing additional *.conf files",
+				},
+				// Int flags
+				&cli.IntFlag{
+					Name:  "test-wait",
+					Usage: "wait up to this many seconds for service inputs to complete in test mode",
+				},
+				//
+				// String flags
+				&cli.StringFlag{
+					Name:  "usage",
+					Usage: "print usage for a plugin, ie, 'telegraf --usage mysql'",
+				},
+				&cli.StringFlag{
+					Name:  "pprof-addr",
+					Usage: "pprof address to listen on, not activate pprof if empty",
```

Comment:
I forgot to click save on this comment...

`, not activate pprof if empty` - this doesn't sound clear enough. Essentially the user has to provide a `host|IP:port`to use for this option, right?

Maybe `pprof host/IP and port to listen on (e.g. 'localhost:8080')`?

Reply:
Seems like a good change to me, I used localhost:6060 as the example because it matches the docs